prov/gni/src/gnix_fabric.c

```diff
@@ -59,6 +59,12 @@
 const char gnix_fab_name[] = "gni";
 const char gnix_dom_name[] = "/sys/class/gni/kgni0";
 
+/* TODO: this will need to be adjustable - probably set in GNI_INI*/
+uint32_t gnix_def_max_nics_per_ptag = 4;
+uint32_t gnix_nics_per_ptag[GNI_PTAG_USER_END] = {0};
```

Comment:
@sungeunchoi I think it depends on whether we expect the list to be mutated concurrently. The solution with a single pthread mutex has less size overhead but means that the entire list is locked at any given time. If we were using an array of atomics that'd give us more flexibility since you'd be able to atomically operate on an element by element basis.